crates/wasmtime/src/runtime/vm/gc/enabled/drc.rs

```diff
@@ -578,9 +566,92 @@ unsafe impl GcHeapObject for VMDrcHeader {
 
 impl VMDrcHeader {
     /// The size of this header's object.
+    #[inline]
     fn object_size(&self) -> usize {
         usize::try_from(self.object_size).unwrap()
     }
+
+    /// Is this object in the over-approximated stack roots list?
+    #[inline]
+    fn is_in_over_approximated_stack_roots(&self) -> bool {
+        self.header.reserved_u27() & wasmtime_environ::drc::HEADER_IN_OVER_APPROX_LIST_BIT != 0
+    }
+
+    /// Is this object in the over-approximated stack roots list?
+    #[inline]
+    fn set_in_over_approximated_stack_roots(&mut self) {
+        debug_assert!(!self.is_in_over_approximated_stack_roots());
+        let reserved = self.header.reserved_u27();
+        self.header
+            .set_reserved_u27(reserved | wasmtime_environ::drc::HEADER_IN_OVER_APPROX_LIST_BIT);
+    }
+
+    /// Remove this object from the over-approximated-stack-roots list.
+    ///
+    /// This does *not* remove the link pointing to this object, if any. That is
+    /// the caller's responsibility.
+    #[inline]
+    fn remove_from_over_approximated_stack_roots(&mut self) {
+        debug_assert!(self.is_in_over_approximated_stack_roots());
+        let reserved = self.header.reserved_u27();
+        self.header
+            .set_reserved_u27(reserved & !wasmtime_environ::drc::HEADER_IN_OVER_APPROX_LIST_BIT);
+        // NB: we don't need to set `self.next_over_approximated_stack_root` to
+        // `None` -- that field only has meaning when the bit is set. This lets
+        // us avoid some unnecessary stores.
+    }
```

Comment:
Could this method be removed entirely in favor taking a `bit: bool` parameter in `set_in_over_approximated_stack_roots` above? That way the callers would inherit the comment of "no need to update the over approximated set" which I think would make more sense in the context of callers than here.